compute/encryption/generate_wrapped_rsa_key.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of authenticating using access tokens directly on Compute Engine.
+
+For more information, see the README.md under /compute.
+"""
+
+# [START all]
+
+import base64
+import os
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+import requests
+
+
+GOOGLE_PUBLIC_CERT_URL = (
+    'https://cloud-certs.storage.googleapis.com/google-cloud-csek-ingress.pem')
+
+
+def get_google_public_cert_key():
+    r = requests.get(GOOGLE_PUBLIC_CERT_URL)
+    r.raise_for_status()
+
+    # Load the certificate.
+    certificate = x509.load_pem_x509_certificate(
+        r.text.encode('utf-8'), default_backend())
+
+    # Get the certicate's public key.
+    public_key = certificate.public_key()
+
+    return public_key
+
+
+def wrap_rsa_key(public_key, private_key_bytes):
+    # Use the Google public key to encrypt the customer private key.
+    # This means that only the Google private key is capable of decrypting
+    # the customer private key.
+    wrapped_key = public_key.encrypt(
+        private_key_bytes,
+        padding.OAEP(
+            mgf=padding.MGF1(algorithm=hashes.SHA1()),
+            algorithm=hashes.SHA1(),
+            label=None))
+    encoded_wrapped_key = base64.b64encode(wrapped_key)
+    return encoded_wrapped_key
+
+
+def main():
```

Comment:
How about to make this script a real utility by just printing out the `wrapperd_rsa_key`, and receiving the csek filename as an argument? That way, customers can use this script like following:

```
cd compute/encryption/
pip install -r requirements.txt
touch /somewhere/my_wrapped_key
chmod 0600 /somewhere/my_wrapped_key
generate_wrapped_rsa_key.py /somewhere/my_private_key > /somewhere/my_wrapped_key
gcloud beta compute instances create example-instance --csek-key-file /somewhere/my_wrapped_key
```

Of course you can write a python script that uses functions in this file, but I think it's easier for some users to just use shell scripting.